### ISSUE TITLE
Cannot copy paste leaf nodes in Chrome

### ISSUE THREAD

**Issue by ahixon** [ProseMirror/prosemirror#884]
### Issue details

Chrome, and possibly Webkit, doesn't seem to differentiate easily between sequential nodes that are atomic (contain no directly editable content). 

Despite the div being entirely selected (either by clicking, or by programatically creating a NodeSelection), trying to copy/paste in Chrome causes no [`copy`](https://developer.mozilla.org/en-US/docs/Web/Events/copy) event to fire within the `contentEditable`.

### Steps to reproduce

https://heather-shape.glitch.me/

https://glitch.com/edit/#!/heather-shape

> In Chrome, select any of the middle "uneditable" nodes and try to copy them.
> 
> Chrome does not generate a copy event. Unsetting contenteditable="false" allows the node to be copyable.
> 
> The end nodes (1 and 4) copy correctly.

As above, trying to create a NodeSelection manually around any of those nodes will create the selection, but any copy keyboard events do nothing.

### ProseMirror version

Versions that are available via the `examples` require.

Also exhibited with:

    "prosemirror-model": "^1.6.1",
    "prosemirror-view": "^1.6.7"

### Affected platforms

- [x] Chrome
- [ ] Firefox
- [ ] Internet Explorer
- [ ] Other

### Possible workaround

Interestingly, changing the `toDOM` to serialise the nodes to an `object` rather than a `div` or similar element allows the node to be selected, copy and pasted correctly.

I'm not sure how cross browser this is, or if this is even a solution for Prosemirror to handle.

**Comment by marijnh** [comment]
Ugh. I'm not sure if the browser is intentionally deciding that copying makes no sense when uneditable content is selected or if it's just bugging out somehow, but yeah, that's awkward.

I haven't been able to find a reasonable fix on the library level yet, but adding either `draggable: true` or `style: "user-select: none"` to the uneditable element works around the issue.

**Comment by marijnh** [comment]
Attached patch might be a reasonable workaround, but I'm not entirely sure it won't have side effects. Could you give it a try and see how well it works for you?

**Comment by ahixon** [comment]
Sorry for the long delay.

Unfortunately this patch doesn't work for me. Enabling/disabling dragging has no effect on copy-ability of these nodes in Chrome. This might be because my usecase renders deeply nested divs inside the NodeView.

We were previously using the `<object>` workaround, but turns out it has layout problems when attempting to float the element.

Setting `contenteditable="true"` has success. Of course, we probably only want to turn this kludge on for Chrome. Other browsers draw the ugly resize handles/exhibit odd behaviour in this case.

I've attached a patch where we turn it on for atomic nodes. Out of interest, why do we explicitly disallow setting the value of contenteditable? Widgets, for instance, already have the (admittedly undocumented) `raw` setting.

**Comment by marijnh** [comment]
Making such nodes editable means that typed text might end up inside them (for example, in Chrome, when they are the only child of their parent textblock that seems to commonly happen). As such, this solution doesn't look very attractive.

**Comment by ahixon** [comment]
Yeah, good point.

Would only modifying the "content-editable" attribute _if it hasn't already been set_ by the nodeview constructor sound reasonable?

**Comment by marijnh** [comment]
That'd be a reasonable (if somewhat obscure) way to opt out of the setting of `contenteditable` to false, I guess. You'd still need to address the issues around typing next to such a node, but if you've explicitly set it to editable it sounds okay for the library to treat that as your own responsibility. I'd be okay with a PR that add this behavior.

**Comment by qinbinhua601** [comment]
@ahixon I happeded to run into this issue too in my project. And I was able to solve this issue by setting one line css to the `.ProseMirror-selectednode[contenteditable="false"]` selectors `-webkit-user-modify: read-write;` (see the red box in the screenshot below for the Steps to reproduce glitch demo):
![2022-05-10 at 08 46](https://user-images.githubusercontent.com/3437641/167520865-944122b8-597f-4ae7-9a31-43eaf3bfcc34.png)

But I don't see the logic here for doing this. Would be appreciate, if anyone could explain more detail for why this worked.

**Comment by marijnh** [comment]
> And I was able to solve this issue by setting one line css

Won't that CSS make the node editable again, though? (With all the problems, such as typed content accidentally ending up somewhere inside it, that entails.)

>  I don't see the logic here for doing this

I'm not sure what 'this' refers to in this sentence.

**Comment by qinbinhua601** [comment]
> Won't that CSS make the node editable again, though? (With all the problems, such as typed content accidentally ending up somewhere inside it, that entails.)
Yes, it's the werid part of the behavior by the browser by setting both `contenteditable="false"` and `-webkit-user-modify: read-write;`. I don't see anything obvious wrong by setting this css, it acted like not setting it in my case. But the copy event really triggered.

> > I don't see the logic here for doing this
> 
> I'm not sure what 'this' refers to in this sentence.
'this' refers to by setting `-webkit-user-modify: read-write;` to .ProseMirror-selectednode[contenteditable="false"] selector which seemed to be a work around for me in my project to this issue.

**Comment by danielleshwed** [comment]
I'm having this same issue. Non content editable nodes will sometimes be copiable - other times the copy event doesnt fire at all. 

it seems worse when there are multiple non editable nodes next to each other vs if there is editable content in between non editable nodes

is there anyone who has discovered a reasonable fix for this?

**Comment by marijnh** [comment]
This issue has been moved to [issue #884](https://code.haverbeke.berlin/prosemirror/prosemirror/issues/884) on the [repository's new home](https://code.haverbeke.berlin/prosemirror/prosemirror/).
